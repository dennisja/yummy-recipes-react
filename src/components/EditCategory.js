import React from 'react';
import { withRouter } from 'react-router-dom';
import AddCategoryForm from './forms/CategoryForm';
import FormCard from './forms/FormCard';

const EditCategory = ({ match }) => {
    const { categoryId } = match.params;
    return (
      <div className="row">
        <div className="col m4">
                side bar component here
        </div>
        <div className="col m8">
          <FormCard
            form={<AddCategoryForm role="edit" categoryId={categoryId} />}
            title="Add New Category"
          />
        </div>
      </div>
    );
};

// const CreateCategoryWithRouter = withRouter(CreateCategory);
export default EditCategory;
