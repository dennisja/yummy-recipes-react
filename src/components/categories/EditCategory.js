import React from 'react';
import PropTypes from 'prop-types';

import AddCategoryForm from '../forms/CategoryForm';
import FormCard from '../forms/FormCard';
import SideBar from '../SideBar';

const EditCategory = (props) => {
    const { categoryId } = props.match.params;
    return (
      <div className="row">
        <div className="col m4">
          <SideBar />
        </div>
        <div className="col m8">
          <FormCard
            form={<AddCategoryForm role="edit" categoryId={categoryId} {...props} />}
            title="Edit Category"
          />
        </div>
      </div>
    );
};

EditCategory.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.object,
    isExact: PropTypes.bool,
    path: PropTypes.string,
    url: PropTypes.string,
  }).isRequired,
};

export default EditCategory;
