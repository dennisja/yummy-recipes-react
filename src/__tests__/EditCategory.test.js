import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import EditCategory from '../components/categories/EditCategory';

describe('Test CreateCategory component', () => {
    const categoryWrapper = shallow(<EditCategory match={{ params: { categoryId: 1 } }} />);
    it('renders properly', () => {
        expect(shallowToJson(categoryWrapper)).toMatchSnapshot();
    });

    it('renders a side bar component', () => {
        expect(categoryWrapper.find('SideBar').length).toBe(1);
    });

    it('renders a form card', () => {
        expect(categoryWrapper.find('FormCard').length).toBe(1);
    });
});
