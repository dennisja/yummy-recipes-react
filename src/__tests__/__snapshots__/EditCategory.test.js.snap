// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CreateCategory component renders properly 1`] = `
<div
  className="row"
>
  <div
    className="col m4"
  >
    <SideBar />
  </div>
  <div
    className="col m8"
  >
    <FormCard
      form={
        <AddCategoryForm
          categoryId={1}
          match={
            Object {
              "params": Object {
                "categoryId": 1,
              },
            }
          }
          role="edit"
        />
      }
      iconClass={null}
      title="Edit Category"
    />
  </div>
</div>
`;
